tests/integration_tests/functional/test_drive_vhost_user.py

```diff
@@ -60,3 +113,182 @@ def test_vhost_user_block(microvm_factory, guest_kernel, rootfs_ubuntu_22):
+
+    # Adding second block device (rootfs)
+    vm.add_vhost_user_drive("rootfs", VHOST_USER_SOCKET, is_root_device=True)
+
+    # Adding third block device.
+    fs2 = drive_tools.FilesystemFile(os.path.join(vm.fsfiles, "scratch2"), size=512)
+    vm.add_drive("scratch2", fs2.path)
+
+    vm.start()
+
+    # Determine the size of the microVM rootfs in bytes.
+    rc, stdout, stderr = utils.run_cmd(
+        "du --apparent-size --block-size=1 {}".format(rootfs_ubuntu_22),
+    )
+    assert rc == 0, f"Failed to get microVM rootfs size: {stderr}"
+
+    assert len(stdout.split()) == 2
+    rootfs_size = stdout.split("\t")[0]
+
+    # The devices were added in this order: fs1, rootfs, fs2.
+    # However, the rootfs is the root device and goes first,
+    # so we expect to see this order: rootfs, fs1, fs2.
+    # The devices are identified by their size.
+    ssh_connection = vm.ssh
+    _check_block_size(ssh_connection, "/dev/vda", rootfs_size)
+    _check_block_size(ssh_connection, "/dev/vdb", fs1.size())
+    _check_block_size(ssh_connection, "/dev/vdc", fs2.size())
+
+
+def test_partuuid_boot(
+    microvm_factory,
+    guest_kernel,
+    rootfs_ubuntu_22,
+    rootfs_ubuntu_22_partuuid_and_disk_path,
+):
+    """
+    This test simply tries to boot a VM with
+    vhost-user-block as a root device.
+    """
+
+    VHOST_USER_SOCKET = "/vub.socket"
+
+    partuuid = rootfs_ubuntu_22_partuuid_and_disk_path[0]
+    disk_path = rootfs_ubuntu_22_partuuid_and_disk_path[1]
+
+    vm = microvm_factory.build(guest_kernel, None, monitor_memory=False)
+
+    # Converting path from tmpfs ("./srv/..") to local
+    # path on the host ("../build/..")
+    rootfs_path = utils.to_local_dir_path(str(rootfs_ubuntu_22))
+    # Launching vhost-user-block backend
+    _backend = spawn_vhost_user_backend(vm, disk_path, VHOST_USER_SOCKET)
+
+    # We need to setup ssh keys manually because we did not specify rootfs
+    # in microvm_factory.build method
+    ssh_key = rootfs_ubuntu_22.with_suffix(".id_rsa")
+    vm.ssh_key = ssh_key
+    vm.spawn()
+    vm.basic_config(add_root_device=False)
+    vm.add_vhost_user_drive(
+        "1", VHOST_USER_SOCKET, is_root_device=True, partuuid=partuuid
+    )
+    vm.add_net_iface()
+    vm.start()
+
+    # Attempt to connect to the VM.
+    # Verify if guest can run commands.
+    exit_code, _, _ = vm.ssh.run("ls")
+    assert exit_code == 0
+
+    # Prepare the input for doing the assertion
+    assert_dict = {}
+    # Keep an array of strings specifying the location where some string
+    # from the output is located.
+    # 1-0 means line 1, column 0.
+    keys_array = ["1-0", "1-6"]
+    # Keep a dictionary where the keys are the location and the values
+    # represent the input to assert against.
+    assert_dict[keys_array[0]] = "ro"
+    assert_dict[keys_array[1]] = "/dev/vda"
+    _check_drives(vm, assert_dict, keys_array)
+
+
+def test_partuuid_update(microvm_factory, guest_kernel, rootfs_ubuntu_22):
```

Comment:
can we add a test where we kill the backend and make sure Firecracker doesn't crash?

Reply:
How can we test that? I assume the VM will just hang in that case.

Reply:
yes, just make sure it does not crash in a weird way. It's fine for it to hang.

Reply:
added